src/containers/MarketPage/MarketPage.js

```diff
@@ -1,23 +1,35 @@
 import React, { useEffect } from 'react';
-import { useDispatch } from 'react-redux';
+import { useDispatch, useSelector } from 'react-redux';
+import CircularProgress from '@material-ui/core/CircularProgress';
 import Transaction from '../Transaction';
 import Table from '../CurrencyTable';
-import { wrapper, table, transaction } from './marketPage.module.scss';
+import { wrapper, table, transaction, spinner } from './marketPage.module.scss';
 import { getTickers } from '../../redux/reducers/stock/actions';
 import { API_CALL_INTERVAL } from '../../utils/constants';
 
+let canShowSpinner;
+
 const MarketPage = () => {
   const dispatch = useDispatch();
+  const { isLoading, stockData } = useSelector((state) => state.stock);
 
   useEffect(() => {
+    canShowSpinner = true;
+
     dispatch(getTickers());
-    const intervalID = setInterval(() => dispatch(getTickers()), API_CALL_INTERVAL);
+    const apiCallIntervalID = setInterval(() => dispatch(getTickers()), API_CALL_INTERVAL);
 
     return () => {
-      clearInterval(intervalID);
+      clearInterval(apiCallIntervalID);
     };
   }, []);
 
+  if ((isLoading && canShowSpinner) || !stockData.length) {
+    return <CircularProgress className={spinner} />;
+  }
```

Comment:
Should I also remove the following checker? Tbh `canShowSpinner` brings nothing because when component rerender after initial api call it will be set on `false` but the data is not yet here.
```suggestion
  if (!stockData.length) {
    return <CircularProgress className={spinner} />;
  }
```

Reply:
Yes, remove that condition.